activerecord/test/cases/associations/has_one_through_associations_test.rb

```diff
@@ -81,11 +81,22 @@ def test_replacing_target_record_deletes_old_association
 
   def test_set_record_to_nil_should_delete_association
     @member.club = nil
+    assert_nil @member.joined_on
+    assert_nil @member.current_membership
+    assert_nil @member.club
     @member.reload
+    assert_nil @member.joined_on
     assert_nil @member.current_membership
     assert_nil @member.club
   end
 
+  def test_set_record_after_delete_association
+    @member.club = nil
+    @member.club = clubs(:moustache_club)
```

Comment:
If `through_proxy` is not reset (remain destroyed (frozen) target), this line causes `RuntimeError: Can't modify frozen hash`.